Xplat/src/main/java/vazkii/botania/common/item/WorldshaperssSextantItem.java

```diff
@@ -219,4 +307,35 @@ public static void render(PoseStack ms, Player player, ItemStack stack) {
 		}
 	}
 
+	@Override
+	public Component getName(@NotNull ItemStack stack) {
+		Component mode = Component.literal(" (")
+				.append(Component.translatable(getModeString(stack)))
+				.append(")");
+		return super.getName(stack).plainCopy().append(mode);
+	}
+
+	public static String getModeString(ItemStack stack) {
+		return "botaniamisc.sextantMode." + (getSphereMode(stack) ? "sphere" : "circle");
+	}
+
+	private static String getRadiusString(double radius) {
+		NumberFormat format = getNumberFormat(1);
+
+		return format.format(radius);
+	}
+
+	// these two methods should probably move to a Locale helper class:
+	private static NumberFormat getNumberFormat(int numDigits) {
+		var format = NumberFormat.getInstance(getLocale());
+		format.setRoundingMode(RoundingMode.HALF_UP);
+		format.setMaximumFractionDigits(numDigits);
+		format.setMinimumFractionDigits(numDigits);
+		return format;
+	}
+
+	private static Locale getLocale() {
+		final String languageCode = Minecraft.getInstance().getLanguageManager().getSelected().getCode();
+		return Locale.forLanguageTag(languageCode);
```

Comment:
this touches client code. I know it's actually safe in both the usages here, but it's easy to introduce easily-missed server-only bugs. I would move getLocale to `Proxy` and have it return en_us by default, then override it in `ClientProxy` to look up the client locale.